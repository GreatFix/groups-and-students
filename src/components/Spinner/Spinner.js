import React from 'react'
import classes from './Spinner.module.css'
import PropTypes from 'prop-types'

const Spinner = (props) => {
  return (
    <div className={classes.Spinner}>
      <div style={{ background: props.centerColor }}></div>
    </div>
  )
}

Spinner.propTypes = {
  centerColor: PropTypes.string,
}

Spinner.defaultProps = {
  centerColor: '#00994f',
}

export default Spinner
