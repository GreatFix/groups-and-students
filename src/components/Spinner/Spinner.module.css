.Spinner {
  display: grid;
  align-content: center;
  justify-content: center;
  width: 10vh;
  height: 10vh;
  position: fixed;
  left: 50%;
  top: 50%;
  background-color: #1e7a6d;
  border-radius: 50%;
  background: linear-gradient(0deg, #025c2f, #0cd34e);
  animation: spin infinite 10s linear;
}
.Spinner div {
  width: 7.5vh;
  height: 7.5vh;
  border-radius: 50%;
  background: #00994f;
}
@keyframes spin {
  from {
    transform: translate(-5vh, -5vh) rotate(0deg);
  }
  to {
    transform: translate(-5vh, -5vh) rotate(360deg);
  }
}
