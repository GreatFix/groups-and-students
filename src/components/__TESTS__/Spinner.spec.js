import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Spinner from '../Spinner/Spinner'

describe('Spinner', () => {
  test('render', () => {
    const { container } = render(<Spinner />)
    expect(container.querySelector('.Spinner')).toMatchSnapshot()
  })
})
