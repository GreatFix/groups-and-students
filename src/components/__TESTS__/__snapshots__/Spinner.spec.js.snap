// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner render 1`] = `
<div
  class="Spinner"
>
  <div
    style="background: rgb(0, 153, 79);"
  />
</div>
`;
